validator/keymanager/v2/derived/derived.go

```diff
@@ -182,14 +183,16 @@ func NewKeymanager(
 		seedCfg:          seedConfig,
 		seed:             seed,
 		accountsPassword: accountsPassword,
-		keysCache:        make(map[[48]byte]bls.SecretKey),
 	}
-	// We initialize a cache of public key -> secret keys
-	// used to retrieve secrets keys for the accounts via the unlocked wallet.
-	// This cache is needed to process Sign requests using a validating public key.
-	if err := k.initializeSecretKeysCache(); err != nil {
-		return nil, errors.Wrap(err, "could not initialize secret keys cache")
+	// Initialize public and secret key caches that are used to speed up the functions
+	// FetchValidatingPublicKeys and Sign
+	k.lock.Lock()
+	err = k.initializeKeysCaches()
```

Comment:
Would rather this initialize keys caches function hold on to the mutex inside of its function body

Reply:
Thinking about it, the initializeKeysCaches function is used only during the initialization of the keymanager. I think it is safe to assume that nobody will try to use the keymanager from a different thread before it is initialized. Therefore the lock around initializeKeysCaches in not really necessary and could be removed.